script.js

```diff
@@ -168,137 +254,121 @@ function addTemp() {
     Active Transport Functions and Globals
 ********************************************/
 
-function toggleAT() {
-    if (activeAT === false) {
-        activeAT = true;
-        spanAT.textContent = "Disable";
+    // Returns a negative or positive number for random movement
+    function randomCo() {
+        // Speed + random value so there's always some movement
+        let distance = speed + (Math.random() * Math.floor(speed));
+        
+        // Picks a direction of negative or positive
+        let direction = (Math.random());
+        if (direction > 0.5) {
+            direction = 1;
+        }
+        else {
+            direction = -1;
+        }
+        
+        return (distance * direction);
     }
-    else {
-        activeAT = false;
-        spanAT.textContent = "Enable";
+
+        // Checks if the x and y coordinates are inside the canvas
+    function isInCanvas(x, y) {
+        if (x > 0 && y > 0 && y < canvasHeight && x < canvasWidth) {
+            return true;
+        } else {
+            return false;
+        }
     }
-    // Update canvas if needed
-    updateDrawing();
-}
 
-// Draws AT, with gradient if AT is enabled
-function drawAT(midpoint, y) {
-    // If AT not active then no gradient
-    let colorAT = "#55066d";
-    // change later to AT 
-    if (activeAT === true) {
-        colorAT = gradientAT;
+class Molecule {
+
+    constructor(moleculeType, startX, startY) {
```

Comment:
`moleculeType` does not seem to properly work. Originally I had it do `this.moleculeType = moleculeType` but `this.moleculeType` stayed at nil, which meant that I couldn't use it for some of the functions like `draw()`. I just pass in the type in those cases but I would like to have the molecule itself store it. LMK if you know how to make it not nil.

Reply:
What are you passing in as a parameter when constructing a new Molecule? You could try creating a moleculeEnum (pick a method from [((https://stackoverflow.com/questions/287903/what-is-the-preferred-syntax-for-defining-enums-in-javascript))], or perhaps just pass in a string and check the strings later when needed?

Reply:
@drichardi Using the enum worked. I had tried before, but I think I used the wrong syntax. This is much cleaner than what I was doing before, thanks!